contracts/core/fee-calculators/PerformanceFeeCalculator.sol

```diff
@@ -0,0 +1,312 @@
+    }
+
+    function getFee()
+        external
+        view
+        returns (uint256)
+    {
+        uint256 streamingFee = calculateStreamingFee();
+
+        uint256 profitFee = 0;
+        if (exceedsProfitFeeFrequency()) {
+            profitFee = calculateProfitFee(streamingFee);
+        }
+
+        return streamingFee.add(profitFee);
+    }
+
+    function updateAndGetFee()
+        external
+        returns (uint256)
+    {
+        uint256 streamingFee = calculateStreamingFee();
+
+        uint256 profitFee = 0;
+        if (exceedsProfitFeeFrequency()) {
+            profitFee = calculateProfitFee(streamingFee);
+        }
+
+        updateFeeState(streamingFee, profitFee);
+
+        return streamingFee.add(profitFee);
+    }
+
+    /* ============ Private Functions ============ */
+
+    function parsePerformanceFeeCallData(
+        bytes memory _callData
+    )
+        private
+        pure
+        returns (InitFeeParameters memory)
+    {
+        InitFeeParameters memory parameters;
+
+        assembly {
+            mstore(parameters,           mload(add(_callData, 32)))   // profitFeeFrequency
+            mstore(add(parameters, 32),  mload(add(_callData, 64)))   // highWatermarkResetFrequency
+            mstore(add(parameters, 64),  mload(add(_callData, 96)))   // profitFeePercentage
+            mstore(add(parameters, 96),  mload(add(_callData, 128)))  // streamingFeePercentage
+        }
+
+        return parameters;
+    }
+
+    function validateFeeParameters(
+        InitFeeParameters memory parameters
+    )
+        private
+        view
+    {
+        require(
+            parameters.profitFeePercentage <= maximumProfitFeePercentage,
+            "PerformanceFeeCalculator.validateFeeParameters: Profit fee exceeds maximum."
+        );
+
+        require(
+            parameters.streamingFeePercentage <= maximumStreamingFeePercentage,
+            "PerformanceFeeCalculator.validateFeeParameters: Streaming fee exceeds maximum."
+        );
+
+        ScaleValidations.validateMultipleOfBasisPoint(parameters.profitFeePercentage);
+        ScaleValidations.validateMultipleOfBasisPoint(parameters.streamingFeePercentage);
+
+        require(
+            parameters.highWatermarkResetFrequency >= parameters.profitFeeFrequency,
+            "PerformanceFeeCalculator.validateFeeParameters: Fee collection frequency must exceed highWatermark reset."
+        );
+    }
+
+    function calculateStreamingFee()
+        internal
+        view
+        returns(uint256)
+    {
+        uint256 timeSinceLastFee = block.timestamp.sub(lastStreamingFeeTimestamp());
+
+        return timeSinceLastFee.mul(streamingFeePercentage()).div(ONE_YEAR_IN_SECONDS);
+    }
+
+    function calculateProfitFee(uint256 _streamingFee)
+        internal
+        view
+        returns(uint256)
+    {
+        uint256 rebalancingSetValue = calculateRebalancingSetValue(msg.sender);
```

Comment:
Random note - I would pass the Set address / msg.sender into this function to enable further flexibility

Reply:
? what are you referencing?

Reply:
Sorry, I would make the rebalancingSet a parameter to this function for extra flexibility.

Reply:
Isn't it already? What function are you talking about?